app/controllers/login_controller.rb

```diff
@@ -43,11 +43,20 @@ def callback
 
     store = Procore::Auth::Stores::Session.new(session: session)
     store.save(token)
-    redirect_to users_home_path
+    redirect_to users_home_path, success: "Access token granted."
+  rescue Procore::Error
+    redirect_to login_index_path, danger: "Something went wrong. Please try again."
```

Comment:
What is `Procore::Error`?

Is it something that could be solved by trying again?

Reply:
It's any error from the RubySDK. If this happens something is probably misconfigured, so trying again wouldn't help. I can remove that part.

Reply:
There's a list of error types here: https://github.com/procore/ruby-sdk#error-handling

I'm going to change it to print out the message, so the user has more information about what went wrong.

Reply:
The message includes more information now:
<img width="841" alt="Screen Shot 2021-03-15 at 12 28 18 AM" src="https://user-images.githubusercontent.com/5005914/111118322-dd6b1a00-8525-11eb-97a9-bbc49b9e0cf5.png">